va/va_test.go

```diff
@@ -92,7 +95,9 @@ func httpSrv(t *testing.T, token string) *httptest.Server {
 	currentToken := defaultToken
 
 	m.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
-		if !strings.HasPrefix(r.Host, "localhost:") && !strings.HasPrefix(r.Host, "other.valid:") {
+		if !strings.HasPrefix(r.Host, "localhost:") &&
+			!strings.HasPrefix(r.Host, "other.valid:") &&
+			!strings.HasPrefix(r.Host, "ipv4.and.ipv6.localhost") {
```

Comment:
This is lacking the `:`. However, rather than add the `:`, it's probably better to do a slight refactor: call SplitHostPort, and then lookup the host in a map of "acceptable hosts."

Or, even better: If you remove the "Bad Host header" check, do tests break? If no tests break, let's remove this check altogether. It creates coupling across distant components that is probably not necessary.